src/NJsonSchema/NJsonSchema.csproj

```diff
@@ -29,7 +29,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Namotion.Reflection" Version="1.0.7" />
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
```

Comment:
Can't we keep v9 to avoid breaking changes and if someone needs to use a newer version he can add it to its application and "overrule" it?

Reply:
@RicoSuter How does that "overruling" work? Do I need to build NJsonSchema myself with the updated Newtonsoft.Json package?

Reply:
You should be able to just install a newer version in your app and it should use this (as you fid). Can you provide the full stack trace.

Reply:
Prerequisites: `OtherProject` requires `Newtonsoft.Json 12.0.2`

#### Given `CurrentProject` requires `Newtonsoft.Json` **9.0.1** and `NJsonSchema` and `OtherProject`
---
Warnings during build:
```
Warning	NU1605	Detected package downgrade: Newtonsoft.Json from 12.0.2 to 9.0.1. Reference the package directly from the project to select a different version. 
 CurrentProject -> OtherProject 1.0.0.0 -> Newtonsoft.Json (>= 12.0.2) 
 CurrentProject -> Newtonsoft.Json (>= 9.0.1)	CurrentProject	C:\CurrentProject\CurrentProject.csproj	1	
```

When running the test which creates an instance of  `OtherProject.Engine`
```
System.TypeInitializationException : The type initializer for 'OtherProject.Engine' threw an exception.
  ----> System.IO.FileLoadException : Could not load file or assembly 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at OtherProject.Engine..ctor(IWorkspace workspace)
   at CurrentProject.JsonSchemaTests.Test() in C:\CurrentProject\JsonSchemaTests.cs:line 35
--FileLoadException
   at OtherProject.Engine..cctor()
```

#### Given `CurrentProject` requires `Newtonsoft.Json` **12.0.2** and `NJsonSchema` and `OtherProject`

When running the test
```
System.IO.FileLoadException : Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at CurrentProject.JsonSchemaTests.Test()
```

-------------

### Solution

I figured out I could solve the issue by requiring `Newtonsoft.Json 12.0.2` in `CurrentProject` and adding the following to `CurrentProject..csproj`
```xml
<Project>
  <!-- ... -->
  <PropertyGroup>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
  </PropertyGroup>
</Project
```